src/bidiMapper/utils/log.ts

```diff
@@ -15,16 +15,18 @@
  */
 export function log(type: string): (...message: any[]) => void {
   return (...messages: any[]) => {
-    const elementId = type + '_log';
+    // If run in browser, add debug message to the page.
+    if (typeof window !== 'undefined' && window?.document?.documentElement) {
```

Comment:
Since this will only run in recent browsers, maybe this simplification?

```suggestion
    if (globalThis?.document?.documentElement) {
```

You could also skip the `window.` bits below. Just accessing `document` directly is more common in my experience.